### ISSUE TITLE
SSR Reflections fade out very early on non-metallic materials

### ISSUE THREAD

**Issue by WickedInsignia** [godotengine/godot#75302]
### Godot version

4.0 Stable

### System information

Windows 11, Nvidia RTX4070ti, AMD Ryzen7700x

### Issue description

Godot's SSR reflections fade out far too early. There is commonly more than enough screen information for a reflection to persist before it fades out into nothing.  This might be a conscious choice to reduce visual artifacts, but every SSR implementation I've seen fades out much later and there's no user-facing option to enforce this behavior in Godot.

In this scene the box reflects nicely onto the plane below:

![Godot_SSRRangeBug01](https://user-images.githubusercontent.com/70243134/227663409-0ed9e9e4-c5fd-4306-935c-e63c62b2ca3a.jpg)

However this is reduced entirely while there is still relevant scene information to retain a reflection:

![Godot_SSRRangeBug02](https://user-images.githubusercontent.com/70243134/227663519-d257a519-2ea6-457d-9280-8f04e0f89429.jpg)

In another more complex example:

![Godot_SSRRangeBug03](https://user-images.githubusercontent.com/70243134/227663569-df66daf2-8540-4ccb-94ca-0329584134ae.jpg)

The reflection is gone entirely once the objects casting them reach the top half of the screen:

![Godot_SSRRangeBug04](https://user-images.githubusercontent.com/70243134/227663660-7fa08fbc-f4d5-43fb-a7bf-06c89ee55111.jpg)

### Steps to reproduce

- Initiate SSR with a large Max Steps count (256 used in the above examples)

### Minimal reproduction project

[SSRShortRange.zip](https://github.com/godotengine/godot/files/11067484/SSRShortRange.zip)

**Comment by Calinou** [comment]
This appears to be related to how metallic the material is. Materials with high metallic don't exhibit this issue:

![Screenshot_20230325_023305](https://user-images.githubusercontent.com/180032/227678257-3d009b99-e04d-4731-81f9-c96428511b5c.png)

![Screenshot_20230325_023259](https://user-images.githubusercontent.com/180032/227678258-469a8fae-7d7c-488a-82ca-c45c77fea6a3.png)

![Screenshot_20230325_023243](https://user-images.githubusercontent.com/180032/227678260-62c65b42-1cf8-41be-b32b-a1ef8dbe840f.png)

![Screenshot_20230325_023236](https://user-images.githubusercontent.com/180032/227678262-95ded74b-1aff-482e-8554-b22787fa7223.png)

**Comment by WickedInsignia** [comment]
Ah yeah that's interesting! Didn't notice that.
The opposite seems to occur there: the fade is extremely abrupt with metallic objects, even with adjustments to Fade Out.

**Comment by Calinou** [comment]
> The opposite seems to occur there: the fade is extremely abrupt with metallic objects, even with adjustments to Fade Out.

The fade is expected to look more abrupt with highly metallic materials, as the reflection can't be shown offscreen.

**Comment by WickedInsignia** [comment]
Of course, but just something I noticed. The fade-out with non-metallic surfaces is less disruptive visually but also occurs far too early. A setting to adjust this fade when reflection information is reduced might help.

**Comment by armmah** [comment]
Seems to be caused by the Schlick term, commenting out the 271 line ends up 'fixing' the issue (see screenshots, both non-metallic).

![HImslR3n7A](https://user-images.githubusercontent.com/31006916/227793661-6b87e8f2-a5b9-4ca0-befd-23a52593d642.png)
![qdjKKK6NSP](https://user-images.githubusercontent.com/31006916/227793662-890f14e6-6c3d-4824-afb4-bd1198fd575e.png)

I'm taking a look to see what's the proper fix that won't introduce other issues.

**Comment by MightiestGoat** [comment]
@armmah I am afraid you are late my friend

**Comment by armmah** [comment]
After some investigation - the Fresnel equation itself (Schlick approx.) is fine, I double checked with reference, and also compared visually with a couple of other open source implementations (frostbite, substance).

So even tho the math checks out (the fresnel term is responsible for damping down the specular energy 90 degree angles between incident ray and surface normal, while on grazing angles (0 degree) the specular is at its maximum), the effect definitely does not look realistic.

Either way I have a hack that changes the curve a bit, it still keeps the general behavior of max at 0 and min at 90, but the curve itself is less aggressive.
![image](https://user-images.githubusercontent.com/31006916/228018265-738dd552-b27c-4e11-96d3-cb0f11ac61b3.png)
(red is before, purple is after).

And here is how it looks in the scene (I added a debug view that splits the screen, left - after, right - before)
https://user-images.githubusercontent.com/31006916/228018850-a7e6370e-b05d-48ad-9cb2-f055401d83b2.mp4

I tested in other test scenes (like godot-demo-projects\3d\material_testers) and it seems to be working fine and closer to the expected look as mentioned in the thread.

I can open a PR but I'm not sure if this is good enough as its not physically accurate (altho the full BRDF of receiving material isn't implemented for SSR either way).
